### PR TITLE
kubectl rolling-update support for same image

### LINKED ISSUE #23497: Kubectl rolling update in 1.2  not behaving the same as 1.1
I have a Kubernetes cluster running version 1.2 and the latest version of kubectl installed. If I perform a rolling update on it the same way when I used the 1.1 client, it fails to update. Is this intended behavior, i.e. do I need to invoke this update differently now or is this a regression/bug ?

Client Info:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.0-alpha.7", GitCommit:"c0fd002fbb25d6a6cd8427d28b8ec78379c354a0", GitTreeState:"clean"}
kServer Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.0", GitCommit:"5cb86ee022267586db386f62781338b0483733b3", GitTreeState:"clean"}
$ kubectl-v1.1.1 version
Client Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.1", GitCommit:"92635e23dfafb2ddc828c8ac6c03c7a7205a84d8", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.0", GitCommit:"5cb86ee022267586db386f62781338b0483733b3", GitTreeState:"clean"}
```

Invoking rolling update with 1.2:
```
$ kubectl --server=http://myclusterurl.com:8080 rolling-update sensu-client --image=sstarcher/sensu:latest
error: Specified --image must be distinct from existing container image
See 'kubectl rolling-update -h' for help and examples.
```

Whereas in 1.1 it worked fine:
```
$ kubectl-v1.1.1 --server=http:/myclusterurl.com:8080 rolling-update sensu-client --image=sstarcher/sensu:latest
Created sensu-client-59ea6198da5cb1fe4df053b249be7f78
Scaling up sensu-client-59ea6198da5cb1fe4df053b249be7f78 from 0 to 3, scaling down sensu-client from 3 to 0 (keep 3 pods available, don't exceed 4 pods)
Scaling sensu-client-59ea6198da5cb1fe4df053b249be7f78 up to 1
Scaling sensu-client down to 2
Scaling sensu-client-59ea6198da5cb1fe4df053b249be7f78 up to 2
Scaling sensu-client down to 1
Scaling sensu-client-59ea6198da5cb1fe4df053b249be7f78 up to 3
Scaling sensu-client down to 0
Update succeeded. Deleting old controller: sensu-client
Renaming sensu-client-59ea6198da5cb1fe4df053b249be7f78 to sensu-client
replicationcontroller "sensu-client" rolling updated
```

### PR DESCRIPTION
**Pull request by j3ffml**
Fixes #23497.

Enables `kubectl rolling-update --image` to the same image, adding a `--image-pull-policy` flag to remove ambiguity. This allows rolling-update to behave as an "update and/or restart" (https://github.com/kubernetes/kubernetes/issues/23497#issuecomment-212349730), or as a forced update when the same tag can mean multiple versions (e.g. `:latest`). cc @janetkuo @nikhiljindal

### CODE REVIEW

**Comment by jlowdermilk** [comment]
ref #23516. This handles that use case safely.

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -262,6 +275,7 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		if err != nil {
 			return err
 		}
+		oldHash += "-orig"
```
Comment (zhouhaibing089) [inline pkg/kubectl/cmd/rollingupdate.go]: maybe give a comment here on why adding this?

Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: Done.

**Comment by janetkuo** [comment]
Suggest add a test for this.

**Comment by nikhiljindal** [comment]
Is there a plan to support the same behavior in deployments as well?

**Comment by jlowdermilk** [comment]
Added test. @nikhiljindal, AFAIK, same-image rolling updates are not being supported in deployments. See https://github.com/kubernetes/kubernetes/issues/23497#issuecomment-212021092

**Comment by jlowdermilk** [comment]
friendly ping

**Comment by roberthbailey** [comment]
@jlowdermilk Please ping once the flaky test issue has been resolved and I'll re-evaluate the cherry pick status.

File: docs/user-guide/kubectl/kubectl_rolling-update.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ kubectl rolling-update frontend-v1 frontend-v2 --rollback
       --dry-run[=false]: If true, print out the changes that would be made, but don't actually make them.
   -f, --filename=[]: Filename or URL to file to use to create the new replication controller.
       --image="": Image to use for upgrading the replication controller. Must be distinct from the existing image (either new image or new image tag).  Can not be used with --filename/-f
```
Comment (osterman) [inline docs/user-guide/kubectl/kubectl_rolling-update.md]: I am confused. Does it still need to be distinct? I thought the point of this PR was to allow rolling-updates of the "same" (e.g. non-distinct) image so long as `--image-pull-policy=always` (as noted below).

**Pull request merged**